lib/command.js

```diff
@@ -47,6 +49,51 @@ function errHandlerOpts (rollbar, context) {
   };
 }
 
+function login () {
+  return new Promise(function (fulfill) {
+    spawn('heroku', ['login'], {stdio: 'inherit'})
+    .on('close', function (error) {
+      if (error === 0) fulfill();
+      else fulfill(login());
```

Comment:
yeah, I think that would be preferable, as it would match the behavior of what we do when someone does not have a token normally, ie: `heroku create` without a valid login attempts a valid login once, and if it fails, it stops trying.